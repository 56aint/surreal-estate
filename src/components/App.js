import React from "react";
import "../styles/App.css";
import { Route, Switch } from "react-router-dom";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import Home from "./Home";

function App() {
  return (
    <div className="App">
      <NavBar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/properties" component={Properties} />
        <Route exact path="/add-property" component={AddProperty} />
      </Switch>
    </div>
  );
}

export default App;
