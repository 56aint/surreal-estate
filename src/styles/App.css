.App {
  text-align: center;
}

body {
  justify-content: center;
  margin: 0px;
  padding: 0px;
  background-color: lightgrey;
}
