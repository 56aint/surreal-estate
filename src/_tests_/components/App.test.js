import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "../../components/App";

describe("App", () => {
  test("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders correctly", () => {
    const { getByText, getByTestId } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const linkElement = getByText(/Aaron Hays/i);
    expect(linkElement).toBeInTheDocument();
    fireEvent.click(getByText(/View Property/i));
    expect(getByText("Price Descending")).toBeInTheDocument();
    expect(getByTestId("sidebar-id")).toBeInTheDocument();
  });
});
