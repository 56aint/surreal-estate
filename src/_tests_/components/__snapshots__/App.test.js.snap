// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
      data-testid="navbar-id"
    >
      <img
        alt="estate-logo"
        src="logo.png"
      />
      <ul
        class="navbar-links"
        data-testid="navbar-links-id"
      >
        <li
          class="navbar-links-item"
          data-testid="navbar-links-item-id"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="navbar-links-item"
          data-testid="navbar-links-item-id"
        >
          <a
            href="/properties"
          >
            View Property
          </a>
        </li>
        <li
          class="navbar-links-item"
          data-testid="navbar-links-item-id"
        >
          <a
            href="/add-property"
          >
            Add Property
          </a>
        </li>
        <li
          class="navbar-links-item"
          data-testid="navbar-links-item-id"
        >
          <a
            href="/favourites"
          >
            Favourites
          </a>
        </li>
        <form
          class="fb-sign-in-form"
          data-testid="fb-sign-in-form-id"
        >
          <h1>
            Something went wrong.
          </h1>
        </form>
      </ul>
    </div>
    <div>
      Home Page
    </div>
    <footer
      class="footer"
    >
      <div
        class="wrapper"
      >
        <div
          class="item"
        >
          <i
            class="fas fa-hand-point-right"
          />
        </div>
        <div
          class="footer-content"
        >
          <p>
            Aaron Hays 2020
          </p>
        </div>
        <div
          class="item"
        >
          <i
            class="fas fa-thumbs-up"
          />
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
